client/app/bundles/course/assessment/question/programming/containers/OnlineEditor/Java/OnlineEditorJavaView.jsx

```diff
@@ -0,0 +1,489 @@
+              expectEquals: <code>{intl.formatMessage(javaTranslations.expectEquals)}</code>,
+            }}
+          />
+        </div>
+        <div style={{ marginBottom: '0.5em' }}>
+          <FormattedMessage
+            id="course.assessment.question.programming.onlineEditorJavaView.testCasesCodeDescription"
+            defaultMessage={
+              '{editor}: Clicking {code} will toggle a code editor for you to write code into each ' +
+              'test case. This allows you to initialize variables and all functions for each test. ' +
+              'For example: {codeExample}' +
+              '{codeExampleExpression} and {codeExampleExpected} can then be input into {expression} and ' +
+              '{expected} respectively.'
+            }
+            values={{
+              expression: <b>{intl.formatMessage(translations.expressionHeader)}</b>,
+              expected: <b>{intl.formatMessage(translations.expectedHeader)}</b>,
+              code: <code>{intl.formatMessage(javaTranslations.testCaseDescriptionCode)}</code>,
+              editor: <code>{intl.formatMessage(javaTranslations.testCaseDescriptionEditor)}</code>,
+              /* eslint-disable react/jsx-indent */
+              codeExample:
+                <pre style={{ marginTop: '0.5em', paddingTop: '3px', paddingBottom: '3px' }}>
+                  <p style={{ marginBottom: 0 }}>
+                    {'int arr [] = {0,0,0}; // Initialize variables'}
+                  </p>
+                  <p style={{ marginBottom: 0 }}>{'addOneToArray(arr); // Make function calls'}</p>
+                  <p style={{ marginBottom: 0 }}>{'int expected [] = {1,1,1}; // Make function calls'}</p>
+                  <p style={{ marginBottom: 0 }}>
+                    {'setAttribute("expression", "addOneToArray([0,0,0])");'}
+                    {' // Override the default expression displayed'}
+                  </p>
+                </pre>,
+              /* eslint-enable react/jsx-indent */
+              codeExampleExpected: <code>{'expected'}</code>,
+              codeExampleExpression: <code>{'arr'}</code>,
+            }}
+          />
+        </div>
+        { errorTextElement }
+        {
+          this.renderTestCases(intl.formatMessage(translations.publicTestCases),
+          testCases, 'public')
+        }
+        {
+          this.renderTestCases(intl.formatMessage(translations.privateTestCases),
+          testCases, 'private')
+        }
+        {
+          this.renderTestCases(intl.formatMessage(translations.evaluationTestCases),
+          testCases, 'evaluation')
+        }
+      </div>
+    );
+  }
+
+  render() {
+    const { intl, autograded, isLoading, hasSubmissions } = this.props;
+    const submitAsFile = this.props.data.get('submit_as_file');
+    let toggleLabel = intl.formatMessage(javaTranslations.submitAsFileToggle);
+    if (hasSubmissions) {
+      toggleLabel = intl.formatMessage(javaTranslations.submitAsFileToggleDisabled);
```

Comment:
The general principle is: i) most of the variable names in code should follow the backend behavior for us to follow easily, and ii) front-end labels can reflect more from the user's perspective (what the student/teacher can do). 

For front-end labels, I would suggest something more intuitive. Off the top of my head, `Template Submission` and `File Submission` sounds clear enough. Additionally as @allenwq suggested, you'll need to add some explanation for each of the modes, and also the limitation on selecting one or the other mode, because the labels do not really tell much.